lib/tenstorrent/bh_arc/CMakeLists.txt

```diff
@@ -95,3 +101,59 @@ set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
 	  ${PYTHON_EXECUTABLE} ${ZEPHYR_CURRENT_MODULE_DIR}/scripts/encode_spirom_bins.py
 	  --board ${PROD_NAME} --build-dir ${CMAKE_BINARY_DIR} --output ${CMAKE_BINARY_DIR}/generated_board_cfg
 )
+
+if(NOT CONFIG_TT_SMC_RECOVERY)
+if(CONFIG_TT_FWTABLE_MOCK)
+  # FIXME: will only work for tt_blackhole
+  set(BOARD_DIR ${ZEPHYR_CURRENT_MODULE_DIR}/boards/tenstorrent/tt_blackhole)
+
+  # Inputs
+  set(FW_TABLE_TXT ${BOARD_DIR}/spirom_data_tables/${PROD_NAME}/fw_table.txt)
+  set(FW_TABLE_PROTO ${SPIROM_PROTOBUFS}/fw_table.proto)
+
+  # Paths
+  set(FW_TABLE_GEN_DIR ${ZEPHYR_BINARY_DIR}/bh_arc)
+  set(FW_TABLE_MOCK_PATH ${FW_TABLE_GEN_DIR}/include)
+  set(FW_TABLE_PYPATH ${FW_TABLE_GEN_DIR}/fw_table)
+
+  # Outputs
+  set(FW_TABLE_MOCK_INC ${FW_TABLE_MOCK_PATH}/tenstorrent/fw_table_mock.inc)
+  set(FW_TABLE_PYLIB ${FW_TABLE_PYPATH}/fw_table_pb2.py)
+
+  # Generate fw_table_pb2.py from fw_table.proto
+  # Note: this is done as part of the build process, not as "extra_post_build_commands"
+  add_custom_command(
```

Comment:
@sbansalTT - it's a bit more verbose, but I wonder if the `add_custom_command`, `generate_unique_target_name_from_filename`, `add_custom_target`, `add_dependencies` flow could be used above where we generate `.py` files from `.proto` files.

Reply:
Probably out of scope of this PR, but just something to consider.